trie/committer.go

```diff
@@ -79,16 +79,29 @@ func (c *committer) commitNeeded(n node) bool {
 	return hash == nil || dirty
 }
 
+// commit collapses a node down into a hash node and inserts it into the database
+func (c *committer) Commit(n node, db *Database) (hashNode, error) {
+	if db == nil {
+		return nil, errors.New("no db provided")
+	}
+	h, err := c.commit(n, db, true)
+	if err != nil {
+		return nil, err
+	}
+	hn, ok := h.(hashNode)
+	if !ok {
+		panic("commit did not hash down to a hashnode!")
+	}
+	return hn, nil
+}
```

Comment:
Myeah, I don't think there's much reason for this to exist. You can move the db check into `commit` and I would not care for the root hash node check (I mean, that's what commit is meant to do, no point in sanity checking that it works).